llpc/context/llpcCompiler.cpp

```diff
@@ -1520,14 +1520,16 @@ Result Compiler::BuildGraphicsPipeline(const GraphicsPipelineBuildInfo *pipeline
   Result cacheResult = Result::ErrorUnknown;
 
   if (!buildingRelocatableElf) {
-    if (m_cache)
+    if (m_cache) {
       cacheResult = lookUpCaches(userCache, &hashId, &elfBin, &cacheEntry);
-    else
+      if (cacheResult == Result::Success)
+        pipelineOut->pipelineCacheAccess = CacheAccessInfo::CacheHit;
+    } else {
       cacheEntryState = lookUpShaderCaches(appCache, &cacheHash, &elfBin, &shaderCache, &hEntry);
-
-    if ((cacheEntryState == ShaderEntryState::Ready) || (m_cache && (cacheResult == Result::Success)))
-      pipelineOut->pipelineCacheAccess = CacheAccessInfo::CacheHit;
-    else
+      if (cacheEntryState == ShaderEntryState::Ready)
+        pipelineOut->pipelineCacheAccess = CacheAccessInfo::InternalCacheHit;
```

Comment:
Same as above, I think that if `appCache` is not nullptr, then it is checking an application cache.

Reply:
Same as above

Reply:
Updated